hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HeapMemStoreLAB.java

```diff
@@ -191,11 +192,18 @@ private Chunk getOrMakeChunk() {
       // No current chunk, so we want to allocate one. We race
       // against other allocators to CAS in an uninitialized chunk
       // (which is cheap to allocate)
-      c = (chunkPool != null) ? chunkPool.getChunk() : new Chunk(chunkSize);
+      if(chunkPool != null) {
```

Comment:
Code and comments still cryptic. Can we make it more explicit? What it takes to change ChunkPool to have a no-preallocation mode?